packages/sdk38/src/lcdapi/staking.spec.ts

```diff
@@ -0,0 +1,508 @@
+      }
+      {
+        const msg: MsgUndelegate = {
+          type: "cosmos-sdk/MsgUndelegate",
+          value: {
+            delegator_address: faucet.address,
+            validator_address: validatorAddress,
+            amount: coin(100, "ustake"),
+          },
+        };
+        const memo = "Test undelegation for wasmd";
+        const { accountNumber, sequence } = await client.getNonce();
+        const signBytes = makeSignBytes([msg], defaultFee, chainId, memo, accountNumber, sequence);
+        const signature = await wallet.sign(faucet.address, signBytes);
+        const tx = {
+          msg: [msg],
+          fee: defaultFee,
+          memo: memo,
+          signatures: [signature],
+        };
+
+        const receipt = await client.postTx(tx);
+        assert(!isPostTxFailure(receipt));
+      }
+
+      await sleep(75); // wait until transactions are indexed
+    }
+  });
+
+  describe("delegatorDelegations", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorDelegations(faucet.address);
+      expect(response).toEqual({
+        height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        result: [
+          {
+            delegator_address: "cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6",
+            validator_address: "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+            shares: jasmine.stringMatching(bigDecimalMatcher),
+            balance: { denom: "ustake", amount: jasmine.stringMatching(nonNegativeIntegerMatcher) },
+          },
+        ],
+      });
+    });
+  });
+
+  describe("delegatorUnbondingDelegations", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorUnbondingDelegations(faucet.address);
+      expect(response.height).toMatch(nonNegativeIntegerMatcher);
+      expect(response.result.length).toEqual(1);
+      expect(response.result[0].delegator_address).toEqual("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6");
+      expect(response.result[0].validator_address).toEqual(
+        "cosmosvaloper1gjvanqxc774u6ed9thj4gpn9gj5zus5u32enqn",
+      );
+      expect(response.result[0].entries.length).toBeGreaterThanOrEqual(1);
+      expect(response.result[0].entries[0]).toEqual({
+        creation_height: jasmine.stringMatching(nonNegativeIntegerMatcher),
+        completion_time: jasmine.stringMatching(dateTimeStampMatcher),
+        initial_balance: "100",
+        balance: "100",
+      });
+    });
+  });
+
+  describe("delegatorTransactions", () => {
+    it("works", async () => {
+      pendingWithoutWasmd();
+      const client = makeStakingClient(wasmd.endpoint);
+      const response = await client.staking.delegatorTransactions(faucet.address);
+      expect(response.length).toEqual(3);
```

Comment:
Is this stable when you run the tests multiple times?

Reply:
Yes but I don't yet understand why.